### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by StauceLukas**
**Summary:**

- Change icons in Login and Register to same width.
- Fixed validation problem when they doubled, added somewhere toastNotification. Added more validations.
- Created 2 modal popups for forgotPassword and ResetPasswordConfirmation.
- Fixed Login reveal eyes.
- Created report bug view, now users can send us email to tell about problem.
- Change UI of All vendors.
- Add mini UI improvements to some views .
- Change a bit code.

Edit:
--------------------------------------------------------------------------------------------------------------------
All vendors: 
allvendors

### CODE REVIEW

**Review by Rangatin — COMMENTED** [review]
Can you show, how the new design changes that you made, look like? It would be very interesting to see.

File: FromLocalsToLocals/Views/Vendors/AllVendors.cshtml [hunk 1]
```diff
@@ -11,23 +11,43 @@
...
+    var itemCountDouble = (!string.IsNullOrEmpty(itemCount)) ? double.Parse(itemCount) : 20;
 }
 
+<head>
+    <style>
+        .card{
+            overflow: hidden;
+            min-height: 400px;
+            box-shadow: 0 0 15px rgba(10, 10, 10, 0.3);
+        }
```
Comment (Rangatin) [inline FromLocalsToLocals/Views/Vendors/AllVendors.cshtml:23]: have you checked how this affects reviews? As they are placed in one big card. And if it has fixed size, it's not good.

If you want do change the format of cards for **one specific purpose** and do not want to implement your changes elsewhere, then I suggest giving a name for your card class, for example:

`class="card myCard"` and then when styling that card you could write:

`card.myCard{...}`

Comment (Rangatin) [reply FromLocalsToLocals/Views/Vendors/AllVendors.cshtml:23]: Also it is a bad practice to include .css styling in your html file when we have site.css dedicated for it.

File: FromLocalsToLocals/Controllers/HomeController.cs [hunk 2]
```diff
@@ -50,5 +61,47 @@ public IActionResult Error()
...
         }
+
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> ReportBug(BugReportVM model)
+        {
+            if (ModelState.IsValid)
+            {
+                if (!string.IsNullOrWhiteSpace(model.TextBug)) { 
+                Execute().Wait();
```
Comment (HabaHaba05) [inline FromLocalsToLocals/Controllers/HomeController.cs:72]: Is there any reason why you have used `Execute.Wait()` not `await Execute`?

File: FromLocalsToLocals/Controllers/AccountController.cs [hunk 3]
```diff
@@ -296,12 +298,42 @@ public async Task<ActionResult> ResetPassword(ResetPasswordVM model)
...
+                if (model.Password.Length >= passwordLength)
+                {
+                    isValid = true;
+                }
+                else
+                {
+                    ModelState.AddModelError("", "Password length must be atleast 6 letters!");
+                    return View();
+                }
+            }
```
Comment (HabaHaba05) [inline FromLocalsToLocals/Controllers/AccountController.cs:313]: Password length can be validated by adding an attribute to your ViewModel property.

Comment (Rangatin) [reply FromLocalsToLocals/Controllers/AccountController.cs:313]: Exactly

File: FromLocalsToLocals/Views/Vendors/AllVendors.cshtml (hunk 1 above)
Comment (StauceLukas) [reply FromLocalsToLocals/Views/Vendors/AllVendors.cshtml:23]: Hmm, i didn't though about it . As I look it did not change reviews card or did and I do not know how it should look like. Also, thank for your tip about site.css I will change it to standard.

File: FromLocalsToLocals/Controllers/HomeController.cs (hunk 2 above)
Comment (StauceLukas) [reply FromLocalsToLocals/Controllers/HomeController.cs:72]: Not particular reason , I found it on example and I stick it to but now I will change it to await !

File: FromLocalsToLocals/Controllers/AccountController.cs (hunk 3 above)
Comment (StauceLukas) [reply FromLocalsToLocals/Controllers/AccountController.cs:313]: Thank you, will change!

File: FromLocalsToLocals/Views/Vendors/AllVendors.cshtml (hunk 1 above)
Comment (Rangatin) [reply FromLocalsToLocals/Views/Vendors/AllVendors.cshtml:23]: As I have noticed, changes in site.css are applied globally so if you want to implement some locally, I believe you can insert `<style>` in your html code.

Comment (StauceLukas) [reply FromLocalsToLocals/Views/Vendors/AllVendors.cshtml:23]: If I am not mistaken I already did it 
```
<style>
        .card {
            overflow: hidden;
            min-height: 450px;
            box-shadow: 0 0 15px rgba(10, 10, 10, 0.3);
        }
</style>
```
If you speaking about something different please comment.

**Pull request merged**